tests/UKHO.ERPFacade.API.FunctionalTests/FunctionalTests/SapCallbackEndpointScenarios.cs

```diff
@@ -0,0 +1,34 @@
+﻿using FluentAssertions;
+using NUnit.Framework;
+using UKHO.ERPFacade.API.FunctionalTests.Service;
+
+namespace UKHO.ERPFacade.API.FunctionalTests.FunctionalTests
+{
+    [TestFixture]
+    public class SapCallbackEndpointScenarios
+    {
+        private SapCallbackEndpoint _sapCallbackEndpoint;
+
+        [SetUp]
+        public void Setup()
+        {
+            _sapCallbackEndpoint = new SapCallbackEndpoint();
+        }
+
+        [Test]
+        public async Task WhenSAPEndpointHitWithNonExistingCorrelationId_ThenSAPShouldReturns404NotFoundResponse()
+        {
+            string payload = "{\r\n\"correlationId\": \"Random-28b3-46ea-b009-5943250a9a42\"\r\n}";
+            var response = await _sapCallbackEndpoint.PostSapCallbackEndPointResponseAsync(payload);
+            response.StatusCode.Should().Be(System.Net.HttpStatusCode.NotFound);
+        }
+
+        [Test]
+        public async Task WhenSAPEndpointHitWithInvalidKey_ThenSAPShouldReturns401UnauthorizedResponse()
```

Comment:
Please correct the test case name. It should be CallbackEndpoint who returns the status codes not SAP.